### ISSUE TITLE
Cannot create secret from principals

### ISSUE THREAD

**Issue by akakakakakaa** [novakov-alexey/krb-operator#118]
Hi, When I use this operator, there are two problems.

1. PrincipalController watch pod by principal name not by labels.

2. In krbServer container, The realm address is set to the krbserver name. 

I changed this
```
[realms]
EXAMPLE.COM = {
   kdc = my-krb
   admin_server = my-krb
   kpasswd_server = my-krb
}
```
  to

```
[realms]
EXAMPLE.COM = {
   kdc = localhost:8888
   admin_server = localhost:8749
   kpasswd_server = localhost:8464
}
```
by referring to the service port. and it's working.

Are there any additional options that need to be set apart from the README description?

**Comment by novakov-alexey** [comment]
@akakakakakaa thanks for interest in this project.
1. Can you give more details what you mean here?
2. krbServer container is using standard Kubernetes routing. That means, KDC, Kadmin and Kpassword processes are exposed via "service" resource. For example: https://github.com/novakov-alexey/krb-operator#service
Yes, it is reusing customer resource name as service name. It should work fine.

**Comment by akakakakakaa** [comment]
Sorry for the lack of explanation.

1. I used two crd in README.md. But not working.
```
apiVersion: krb-operator.novakov-alexey.github.io/v1
kind: KrbServer
metadata:
  name: my-krb
  namespace: test
spec:
  realm: EXAMPLE.COM  
```

<pre>
apiVersion: krb-operator.novakov-alexey.github.io/v1
kind: Principals
metadata:
  name: <b>my-krb1</b>
  namespace: test
  labels:
    krb-operator.novakov-alexey.github.io/server: my-krb # reference to KrbServer
spec:
  list:
    - name: client1
      password:
        type: static
        value: mypass
      keytab: cluster.keytab
      secret:
        type: Keytab
        name: cluster-keytab
    - name: user2
      keytab: cluster.keytab
      secret:
        type: KeytabAndPassword
        name: cluster-keytab
</pre>

If i changed name to my-krb, it's working

<pre>
apiVersion: krb-operator.novakov-alexey.github.io/v1
kind: Principals
metadata:
  name: <b>my-krb</b>
  namespace: test
  labels:
    krb-operator.novakov-alexey.github.io/server: my-krb # reference to KrbServer
spec:
  list:
    - name: client1
      password:
        type: static
        value: mypass
      keytab: cluster.keytab
      secret:
        type: Keytab
        name: cluster-keytab
    - name: user2
      keytab: cluster.keytab
      secret:
        type: KeytabAndPassword
        name: cluster-keytab
</pre>

In my guess, third option in pods.waitForPod needs to be changed context.meta.name
https://github.com/novakov-alexey/krb-operator/blob/12e12936554b65829748af65b256d175519e1550/src/main/scala/io/github/novakovalexey/krboperator/service/Kadmin.scala#L80-L92

to serverName
https://github.com/novakov-alexey/krb-operator/blob/12e12936554b65829748af65b256d175519e1550/src/main/scala/io/github/novakovalexey/krboperator/PrincipalsController.scala#L66

2. is kubernetes pod can detect ip by service?

When I installed krbserver, service type is clusterIP.

If you intend to use Local DNS through a kube proxy, I know it should be created as a headless service.
https://kubernetes.io/docs/concepts/services-networking/service/#headless-services

and, port number should be provided.

[realms]
EXAMPLE.COM = {
   kdc = my-krb:8888
   admin_server = my-krb:8749
   kpasswd_server = my-krb:8464
}

If there is anything I missed, I would appreciate it if you could explain it.

**Comment by novakov-alexey** [comment]
1. Nice catch. I have just fixed this bug in version 0.4.18
```bash
docker pull alexeyn/kerberos-operator:0.4.18
```
can you please test it?

2. KDC and other host are availbale via Kubernetes service, i.e. another Pod, for example Kafka Pod can access KDC via my-krb:8888. If you set `localhost` to all Kerberos services, then it may not succeed during the principal authentication, because client side may raise an error like: server hostname is expected to be KrbServer.metadata.name but was `localhost`. Still, I am not sure here, it may still work, if Kubernetes swaps service hostname with `KrbServer.metadata.name` during the Kerberos authentication (client to server handshake), then it may succeed.

**Comment by akakakakakaa** [comment]
Thank you for your reply.

1. using 0.4.18 version, working well.

2.  Sorry for my mistake. It's my kube proxy problem.
 
Thank you for your support. :)
